pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -297,12 +297,12 @@ func (m *Multicluster) ClusterDeleted(clusterID cluster.ID) error {
 		log.Infof("cluster %s does not exist, maybe caused by invalid kubeconfig", clusterID)
 		return nil
 	}
-	if err := kc.Cleanup(); err != nil {
-		log.Warnf("failed cleaning up services in %s: %v", clusterID, err)
-	}
 	if kc.workloadEntryStore != nil {
 		m.opts.MeshServiceController.DeleteRegistry(clusterID, provider.External)
 	}
+	if err := kc.Cleanup(); err != nil {
```

Comment:
Does this order matter?

Reply:
Not really, as long as this is called _after_ the stopCh for `kc` is closed.

Reply:
We could actually make it so `Cleanup` isn't an exported method, and instead schedule it from within `Run`.